code/game/objects/items/weapons/tanks/watertank.dm

```diff
@@ -128,16 +128,16 @@
 
 	var/obj/item/watertank/tank
 
-/obj/item/reagent_containers/spray/mister/New(parent_tank)
-	..()
-	if(check_tank_exists(parent_tank, src))
-		tank = parent_tank
-		reagents = tank.reagents	//This mister is really just a proxy for the tank's reagents
-		loc = tank
-	return
+/obj/item/reagent_containers/spray/mister/Initialize(mapload)
+	if(!check_tank_exists(loc, src))
+		return INITIALIZE_HINT_QDEL
+	tank = loc
+	reagents = tank.reagents	//This mister is really just a proxy for the tank's reagents
+	return ..()
 
 /obj/item/reagent_containers/spray/mister/Destroy()
 	tank = null
+	reagents = null // Unset, this is the tanks reagents
```

Comment:
Fun fact. This went okay because of the order of deletion. Tank -> Nozzle -> Reagents which unsets the tanks reagents var.